main/menu.js

```diff
@@ -113,6 +117,21 @@ function showOpenTorrentFile () {
   })
 }
 
+// Open the preferences window
+function showPreferencesWindow () {
+  windows.createPreferencesWindow()
+}
+
+function showSelectDownloadDirectory() {
+  electron.dialog.showOpenDialog(windows.preferences, {
+    title: 'Select a directory to save in.',
```

Comment:
This looks like the right place for `showSelectDownloadDirectory`. I'd delete the other copy of this function down in `renderer/preferences.js`